### ISSUE TITLE
Allow followpoints to appear on a new combo

### ISSUE THREAD

**Issue by Snowmander** [ppy/osu#6148]
## Proposal:
Currently, when starting a new combo, followpoints do not appear (1 – 2 – 3   1 – 2 – 3). It is my proposal that followpoints should appear regardless of combos (1 – 2 – 3 – 1 – 2 – 3).

This is a gameplay altering change and certainly requires a discussion.

(Could become a skin option.)

## Why does osu! behave like this?
Because the original game (ouendan) did it this way and osu! started out as a PC port.

 ![ouendan](https://user-images.githubusercontent.com/45163894/65081997-de2df900-d9df-11e9-982f-c1fc6a2d1ccc.PNG)

osu! used to be slow-paced compared to today and combos used to serve a purpose of separating the patterns. Each pattern used to correlate to a verse, and a matching dance move. The original game and early osu! could justify this inconsistency for the following reasons; the maps were low AR / star (3-4*), and the different color of the new combo increased the visibility, making it clear where to go next once all the current colored circles disappeared. This is still somewhat true to this day, but as the difficulty / AR increase, this mechanic begins to fall apart. 

## How is this a problem today?

What makes osu! different from its predecessor is its almost infinite number of maps and difficulty. This has allowed the players to focus on a different skillset than ouendan, which is twitch reaction. As the game evolved, followpoints evolved as well into something more like a _followline_. What used to be dots that existed only because counting hitnumbers was hard has become a mechanic for twitch reaction.

   
![1](https://user-images.githubusercontent.com/45163894/65082084-0fa6c480-d9e0-11e9-9a24-926c2cdce77e.png)
![2](https://user-images.githubusercontent.com/45163894/65082087-12a1b500-d9e0-11e9-8eae-049a2159d9c7.png)

These are ranked maps ([1](https://osu.ppy.sh/beatmapsets/440650#osu/959426), [2](https://osu.ppy.sh/beatmapsets/332532#osu/736215)). Same on stable.

It is important that a game should not ask for a superfluous mechanic, and I believe that in osu!, the memorization of a map should not be part of the expected skillset. Of course, it will help to have memorized the map, but the game should treat sightreaders (people playing the map for the first time) fairly. 

It could be argued that the absence of followpoints is actually a part of osu!’s mechanic. But, if followpoints are something that can be _randomly absent_, I think for the consistency’s sake, all followpoints in the game should be removed, and that is probably not going to happen.

The criticism and frustration of the users towards maps with sudden changes to slider velocity is the proof that players expect osu! to not be dependent on map memorization.

To my joy, the game is already facing against memorization-based mechanics. The example of this are spinners. Even on the original game, spinners used to come out of nowhere and catch sightreaders by a surprise. Spinners on lazer have an indicator beforehand, and players don’t have to constantly try to predict a spinner. (We all experience that gut feeling and anxiety of an incoming spinner based on how the music is going.) I take this as a sign that osu! is continuing to evolve and smoothing out its rough edges.

Thank you for reading my proposal.

**Tl;dr** Followpoints now serve a different and important purpose of aiding twitch reaction, and resolving their inconsistency will make them a mechanic that players can rely on.

## **Potential problems with this change:**

Legacy scores with visibility reduction mods (FL and HD).

**Comment by peppy** [comment]
Ranking criteria should disallow such short combos, in my opinion. The follow points not being between combos makes sense, since (at least when i was somewhat involved in the ranking criteria) new combos also meant changes in distance spacing, where follow points may become a distraction by overlapping/intersecting.

**Comment by MightyHelper** [comment]
Since I posted #6778(and got referenced to #6214), I feel that fewer followpoints makes the map harder.
I suggest if this is added, to make it a mod or allow the beatmap creator to decide since, in my opinion, it makes the game slightly easier.
I like 12's not having the extra line, it seems neater to me.

**Comment by Snowmander** [comment]
I don’t understand what you’re talking about with #6214 as it’s about missing followpoints due to a bug and this is a feature suggestion.

I’m specifically against leaving it up to the map creators as it would further complicate building the muscle memory. Also, for the reasons above, I would like to see this change available to ranked as a visual setting, rather than a mod.

Not sure what you mean with “12’s having an extra line”. If you don’t like followpoints, you can turn them off.

**Comment by MightyHelper** [comment]
I meant it in the way of: I've felt almost having no follow points on maps and it feels a lot harder, so adding the extra one seems to make the map slightly easier (when compared to before)

I completely agree with your idea of letting the mapper choose though.

**Comment by peppy** [comment]
Combo points on new combo is going against the whole point of combos, which is to split up passages. Large jumps should be in the same combo if follow points are to be shown to the user. Switching to a new combo generally suggests moving to a new location on the screen to start a new passage, which would get too busy if follow points were shown.

They aren't missing "because ouendan", but because it *makes sense*.

I believe this should be up to mapping criteria, not game mechanics.

**Comment by Snowmander** [comment]
I agree that new combos should be used for splitting up the passages. However, there seems to be an unspoken limit on how big a combo can get (usually below 10) and mappers feel obligated to artificially break up a combo, resulting in situations where a similar consecutive jump would not get followpoints. 

If the purpose of a combo is to separate passages, maybe relegating that role to the difference in combo color could be enough. Followpoints could then be considered an aiming and map reading mechanic, instead of a combo mechanic.

> Switching to a new combo generally suggests moving to a new location on the screen to start a new passage, which would get too **busy** if follow points were shown.

Those still are a challenging jumps especially in faster and harder songs. Precisely the busyness a player need to guide the cursor movement. At the end of the day, players just want to be able to move the cursor from point A to B, irrelevant to the splitting of the combos.

And then there’s the problem of not being able to correct already submitted beatmaps. I still feel confident that in most modern maps above a certain difficulty (5*+) make more sense with having followpoints between combos as the combos become mostly meaningless in higher difficulties.

I don’t have much more to contribute regarding this issue so this will be my last comment. Whether you allow this option to exist or not, the game is still enjoyable and I appreciate your dedication.

**Comment by ebolite** [comment]
In my opinion adding this as a skin option or in the options menu would be an amazing quality of life change for FL players. At higher levels FL is virtually impossible to sightread due to new combos appearing quickly beyond the flashlight FOV, and because there are no follow points between them, the only way to know where the circle is going to be is by repeating the map over and over again. 

What is worth considering is the fact that this would indisputably make FL mod easier for those who decide to enable the option. This could be a bad thing, since it partially invalidates earlier FL plays, but it could also be a good thing, because currently FL is the only difficulty-increasing mod that is completely non-viable at higher level play. 

Adding the option for follow points to span between new combos would open up FL to a wider audience by making it less frustrating and more consistent to play. While it would still rely partially on memorization, it could open the way to more interesting FL gameplay. As a FL player myself, I would love to see this change implemented.

**Comment by smoogipoo** [comment]
This is something that needs further community discussion, and is better suited to the forums: https://osu.ppy.sh/community/forums/13

As for lazer, this isn't something that's going to be worked on by the core team until after the game's finished, so I will be closing it for now.
